web/src/js/ducks/options.js

```diff
@@ -21,12 +22,26 @@ export default function reducer(state = defaultState, action) {
                 ...action.data,
             }
 
+        case ERROR:
+            return {
+                ...state,
+                ...action.data,
+            }
+
         default:
             return state
     }
 }
 
 export function update(options) {
-    fetchApi.put('/options', options)
-    return { type: REQUEST_UPDATE }
+    let error = ''
+    fetchApi.put('/options', options).then(
+        (response) => {
+            response.text().then(errorMsg => {
+                error = errorMsg
+                console.log(error)
+            })
+        }
+    )
+    return {type: ERROR, error}
```

Comment:
we could send the update request to the backend in the way, but there are two problems:
1. the statement in the promise could not change the value of error.
2. even we connect this in the component, the reducer could not receive this action, therefore the error msg could not be stored in the state.
I'm quite confusing about this, how should we fix this?